src/Chunk.ts

```diff
@@ -570,7 +564,7 @@ export default class Chunk {
 
 		// populate ids in the rendered declarations only here
 		// as chunk ids known only after prerender
-		for (const dependency of this.dependencies) {
+		for (const dependency of this.allDependencies) {
```

Comment:
This `allDependencies` isn't ideal, but requires a deeper refactoring to handle any more direct binding system.